### ISSUE TITLE
`torch.compile` (inductor) silently accepts `logcumsumexp` on a 0-d uint8 tensor that eager rejects

### ISSUE THREAD

**Issue by tinywisdom** [pytorch/pytorch#197807]
### 🐛 Describe the bug

`torch.logcumsumexp` is not implemented for the `uint8` (`Byte`) dtype — eager raises
`NotImplementedError`. Under `torch.compile` (inductor), a **0-dimensional** `uint8` input is
**silently accepted** (returned unchanged) instead of raising, because inductor lowers the
0-d cumulative op to an identity/copy that never reaches the dtype-checked kernel. A 1-D `uint8`
input is correctly rejected by both. So the compiled path drops the dtype validation, but only
on the 0-d fast path.

## Minimal repro

```python
import torch

def fn(a):
    return torch.logcumsumexp(a, dim=0)

x = torch.tensor(0, dtype=torch.uint8)          # 0-d (scalar) uint8

fn(x)                                            # eager:    NotImplementedError ("... not implemented for 'Byte'")
torch.compile(fn, backend="inductor")(x)         # compiled: OK -> tensor(0, dtype=torch.uint8)   (silently accepted)
```

Scope (what does / doesn't diverge):

| input | eager | compiled |
|---|---|---|
| **0-d `uint8`** | NotImplementedError (Byte) | **accepts -> tensor(0, uint8)** |
| 1-D `uint8` | NotImplementedError (Byte) | NotImplementedError (Byte) — consistent |
| 0-d `float32` | OK (`1.5`) | OK — consistent |
| 1-D `float32` | OK | OK — consistent |

So the divergence is specific to **0-d + an unsupported dtype**: inductor's scalar shortcut
skips the dtype check.

## Expected vs actual

- **Expected**: compiled and eager agree — either both reject `uint8` or both accept it.
- **Actual**: eager raises `NotImplementedError` for `Byte`; inductor's 0-d path returns the
  input unchanged, so a program that errors in eager silently "works" once compiled.

### Versions

```
PyTorch version: 2.15.0a0+gite67e43b
Is debug build: True
CUDA used to build PyTorch: 12.6
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.4 LTS (x86_64)
GCC version: (Ubuntu 12.3.0-1ubuntu1~22.04.3) 12.3.0
Clang version: 17.0.6
CMake version: version 4.0.0
Libc version: glibc-2.35

Python version: 3.12.13 | packaged by conda-forge (64-bit runtime)
Python platform: Linux-6.8.0-138-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 12.6.20
GPU 0: NVIDIA RTX A6000
Nvidia driver version: 580.82.07
cuDNN version: 9.3.0

[pip3] numpy==2.3.5
[pip3] torch==2.15.0a0+gite67e43b
[pip3] triton==3.7.1
```

cc @malfet @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

**Comment by greekera1000** [comment]
hello @tinywisdom , @malfet  I've opened a fix in #197808 ,check it out whenver u have  time 

The `logcumsumexp` meta function only checked `dim`, not dtype, so tracing accepted integer/bool inputs. On CPU, the 0-d fast path then returned the input unchanged. On CUDA, 1-D integer inputs crashed during Triton compilation instead of raising cleanly. The PR adds a floating/complex dtype check to the meta function, so compiled now raises `NotImplementedError` wherever eager does (verified on a T4: 4 mismatches before, 0 after).
